concent_api/core/message_handlers.py

```diff
@@ -1417,3 +1428,17 @@ def is_message_recieved_in_wrong_state(subtask_id, forbidden_states):
 
 def are_items_unique(items: list):
     return len(items) == len(set(items))
+
+
+def are_eth_addresses_and_singularities_equal_in_all_subtask_results_accepted_list(force_payment: ForcePayment) -> bool:
+    subtask_results_accepted_list = force_payment.subtask_results_accepted_list
+
+    first_key = subtask_results_accepted_list[0].task_to_compute.requestor_ethereum_public_key
+    first_address = subtask_results_accepted_list[0].task_to_compute.requestor_ethereum_address
+
+    for subtask_results_accepted in subtask_results_accepted_list:
+        if first_key != subtask_results_accepted.task_to_compute.requestor_ethereum_public_key:
+            return False
+        if first_address != subtask_results_accepted.task_to_compute.requestor_ethereum_address:
+            return False
+    return True
```

Comment:
This function should be placed in `core/utils.py` not in `core/message_handler.py`. I remember that, there are more functions like this that can be removed from this file and transferred to `utils`. It might be another issue and be done in new PR, but this function should be placed there right now.

Reply:
done